packages/axelar-wasm-std/src/voting.rs

```diff
@@ -320,18 +330,23 @@ impl WeightedPoll {
         let consensus_participants = self
             .participation
             .iter()
-            .filter_map(|(address, participation)| {
-                participation.vote.as_ref().and_then(|votes| {
-                    let voted_consensus = votes.iter().zip(results.iter()).all(|(vote, result)| {
-                        result.is_none() || Some(vote) == result.as_ref() // if there was no consensus, we don't care about the vote
-                    });
-
-                    if voted_consensus {
-                        Some(address.to_owned())
-                    } else {
-                        None
-                    }
-                })
+            .filter_map(|(address, _)| {
+                voting_history
```

Comment:
voting_history is just used as a lookup, yet it's iterated for each address. convert it into a Hashmap first and do constant time lookup in the filter_map

Reply:
I feel like Hashmap will be more readable, so I'll switch to Hashmap. But I am not sure that it's actually more performant: oftentimes linear searching small vectors like this is actually faster than lookup in a hashmap (because all the memory is contiguous in a vector and so searching is very cache friendly), and the hashmap lookup only becomes faster once the dataset grows larger.